tests/integ/scala/test_query_tag_suite.py

```diff
@@ -30,36 +32,47 @@ def test_set_query_tag(session, query_tag):
     try:
         session.query_tag = query_tag
         assert session.query_tag == query_tag
+        remote_query_tag = get_remote_query_tag(session)
+        assert remote_query_tag == query_tag
+        session.query_tag = None
+        remote_query_tag = get_remote_query_tag(session)
+        assert remote_query_tag == ""
+        assert session.query_tag is None
     finally:
         Utils.unset_query_tag(session)
 
 
-@pytest.mark.xfail(reason="SNOW-575699 flaky test", strict=False)
 def test_query_tags_in_session(session):
     query_tag = Utils.random_name_for_temp_object(TempObjectType.QUERY_TAG)
     view_name = Utils.random_name_for_temp_object(TempObjectType.VIEW)
     temp_view_name = Utils.random_name_for_temp_object(TempObjectType.VIEW)
     try:
         session.query_tag = query_tag
-        session.create_dataframe(["a", "b", "c"]).collect()
-        session.create_dataframe(["a", "b", "c"]).count()
-        session.create_dataframe(["a", "b", "c"]).show()
-        session.create_dataframe(["a", "b", "c"]).first()
-        session.create_dataframe(["a", "b", "c"]).to_pandas()
-        session.create_dataframe(["a", "b", "c"]).create_or_replace_temp_view(view_name)
-        session.create_dataframe(["a", "b", "c"]).create_or_replace_view(temp_view_name)
-        query_history = get_query_history_for_tags(session, query_tag)
+        with session.query_history() as query_history:
+            session.create_dataframe(["a", "b", "c"]).collect()
+            session.create_dataframe(["a", "b", "c"]).count()
+            session.create_dataframe(["a", "b", "c"]).show()
+            session.create_dataframe(["a", "b", "c"]).first()
+            if installed_pandas:
+                session.create_dataframe(["a", "b", "c"]).to_pandas()
+            session.create_dataframe(["a", "b", "c"]).create_or_replace_temp_view(
+                view_name
+            )
+            session.create_dataframe(["a", "b", "c"]).create_or_replace_view(
+                temp_view_name
+            )
         Utils.drop_view(session, view_name)
         Utils.drop_view(session, temp_view_name)
 
         assert (
-            len(query_history) == 8
-        )  # 7 DataFrame queries + 1 query that get sql history
+            len(get_local_query_tags(query_history)) == 0
+        )  # The session has query tag. Each SQL statement shouldn't have its own query tag in the statement params.
```

Comment:
We probably should also check session level param? Are we able to do that? Otherwise this test does not seem very useful because we could get the same result by not setting any tags.